Spec/RealtimeClientConnection.swift

```diff
@@ -3028,26 +2989,165 @@ class RealtimeClientConnection: QuickSpec {
                     default:
                         fail("unhandled: \(fixtureMessage["expectedType"].string!)")
                     }
+                }
 
-                    waitUntil(timeout: testTimeout) { done in
-                        channel.publish([message]) { err in
-                            if let err = err {
-                                fail("\(err)")
+                // https://github.com/ably/wiki/issues/22
+                it("should encode and decode fixture messages as expected") {
+                    let options = AblyTests.commonAppSetup()
```

Comment:
> The point of our interoperability tests was to check that a message when decoded/encoded via JSON or MsgPack are consistent with the types we expect to see.

That's the point of the second interoperability test. The point of this one is to check it the message _data_ (not the message itself) when decoded/encoded as string, JSON or base64. We need both precisely because they test two completely independent things.

>  We agreed the only reliable way to get data into the Ably system was via a raw REST request to Ably, and then to check that it was as expected, was also via a raw REST history request to Ably and then to parse the JSON

That part I agree with, and if that's what you meant then OK. But since I'm doing a direct raw POST with a JSON payload ([here](https://github.com/ably/ably-ios/pull/459/files/3176c628c244243510171e5852d62780c41df921..1ee851cb4020e66c1a893573bd687aca9076f0a2#diff-8c3eccfb7635c28b9fadd0e4fcb6e6b0R3019)) it doesn't matter what I set `useBinaryProtocol` to. Same for the raw GET for history. It still doesn't affect how the message gets transmitted via realtime.

> This test uses MsgPack, so the JSON encoding / decoding of a message injected or received via a raw request is not being done.

Yes it is, because what we're encoding/decoding is the JSON _inside_ that message, which comes as a string. [These lines](https://github.com/ably/ably-ios/blob/master/Source/ARTDataEncoder.m#L142-L149) that decode the JSON from the message data are being executed both if the message came as JSON or as MsgPack. At that point, it's just an `ARTMessage`.